### DISCUSSION TITLE
Simultaneously different change events?

### DISCUSSION THREAD

**Discussion by Ademsk1** [xyflow/xyflow#4631] [Q&A]
Hi folks

I noticed in the docs that the function `onNodesChange` has the `changes` argument of type `NodeChange[]` , and `NodeChange` is a union type of selection, deletion, add, dimension etc.

I've also noticed that when I've made selection and deletion changes on the flow, the `changes` array is always an array of one particular type (select, dimensions, delete etc)

I now need to filter for particular changes in order to display different results i.e. a modal display if the user is certain they'd like to delete a node. But when the above came to my attention I wondered... do I need to filter the changes array if I know that the changes array has values of the same type? Can I just check the first element and say, ok, it's all select / delete, perform this action?

Hopefully this makes sense. I'm pretty certain from my basic knowledge of event listeners this will be the case, but wanted to double check. If this is the case, I'd like to change the docs type description to reflect this.

Cheers!

**Comment by bcakmakoglu** [comment]
If you're using the `NodeResizer` you can receive changes of `dimensions` *and* `position`.
In other cases it's probably just one type of change.

Although if you're just trying to do a confirmation before deleting, you can use the [`onBeforeDelete`](https://reactflow.dev/api-reference/react-flow#on-before-delete) prop.
